Notebook: Front Page DeepExplainer MNIST Example

LakshmanKishore opened the pull request:
## Overview

Closes #2819 , #1343  and related task in #3036
Description of the changes proposed in this pull request:

The original notebook contained an example referenced from this [URL](https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py). However, upon visiting the URL, I found that the file was missing. Instead, it had been updated with a new file containing recent code functionality from Keras. I have accordingly updated the example in this notebook with the recent code and re-executed all the cells.


## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
~~Unit tests added (if fixing a bug or adding a new feature)~~

LakshmanKishore committed:
Update Notebook: Front Page DeepExplainer MNIST Example

CloseChoice commented on notebooks/image_examples/image_classification/Front Page DeepExplainer MNIST Example.ipynb:
```diff
@@ -229,46 +136,22 @@
      "text": [
-      "Using TensorFlow backend.\n",
-      "keras is no longer supported, please use tf.keras instead.\n"
+      "IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
```
You have a couple warnings here. Can you please update juypter and ipywidgets to remove these?

LakshmanKishore replied:
Yes, these warnings got removed after updating the jupyter and ipywidgets.

CloseChoice commented on notebooks/image_examples/image_classification/Front Page DeepExplainer MNIST Example.ipynb:
```diff
@@ -17,206 +17,113 @@
    "outputs": [
     {
-     "name": "stdout",
+     "name": "stderr",
```
Can you please remove the `stderr` output here?

CloseChoice commented on notebooks/image_examples/image_classification/Front Page DeepExplainer MNIST Example.ipynb:
```diff
@@ -290,58 +173,31 @@
-   "outputs": [
-    {
-     "data": {
-      "image/png": "iVBORw0KGgoAAAANSUhEUgAACOgAAAMwCAYAAACQ783iAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAWJQAAFiUBSVIk8AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAIABJREFUeJzs3WmUJGd95/tf7lvtS3erN7UWaO1IwkIsBtlgMFh4BFzhxiMZHw82xofF18uMwTPXPmZ0PHiM4ZwB7hgfI/vey2BjY/AgGZCQLLFYgNRaW1K3pN7X6q6uytqyMjMyI/K+eDIi84nIWrq7Ut1d/f286YzIJyKj+sQ/nyXj+T+xRqMhAAAAAAA ...
```
shouldn't this be 5 predictions, since you changed the slicing from `[1:5]` to `[0:5]`

LakshmanKishore replied:
Yes, you are right It should be 5. I will update this in the new commit.

CloseChoice commented:
@LakshmanKishore, thanks a lot for the PR and I hope we can merge this quickly.
Which tensorflow version are you using? I ran your notebook with `2.15.0` and got quite a lot of warnings about eager mode and some deprecations. Would be great if we could fix these too.

LakshmanKishore commented:
This notebook's cells was executed with tensorflow `2.14.0` version. 
Now I have updated the tensorflow to `2.15.0` - latest version and executed the cells. I'm able to still see the warnings.
To remove those warnings we can add this code in the cell
`tf.compat.v1.keras.backend.set_learning_phase(0)` and the warnings will be removed. 
Should I need to go for this?


PS: When I added the above code and ran the cells all the warnings were gone, but when I restarted the kernel and ran those cells the warnings were back, so checking it out.

LakshmanKishore committed:
Merge branch 'shap:master' into notebook-Front-Page-DeepExplainer-MNIST-Example

LakshmanKishore committed:
Merge branch 'master' into notebook-Front-Page-DeepExplainer-MNIST-Example

LakshmanKishore committed:
Merge branch 'master' into notebook-Front-Page-DeepExplainer-MNIST-Example

connortann committed:
Merge branch 'master' into notebook-Front-Page-DeepExplainer-MNIST-Example

connortann committed:
Remove blank cells

connortann committed:
Remove stderr

connortann reviewed (approved):
I pushed a few commits to touch up the minor issues above, LGTM

LakshmanKishore commented:
Hi @connortann 
I was curious to know on how were you able to remove those errors?

connortann commented:
@LakshmanKishore I just manually removed the "stderr" sections from the JSON file using a text editor.

LakshmanKishore commented:
I can see in #3460 we are having a automated way of running the notebooks. 
So while we run via this, I guess the stderr's might appear again.

connortann commented:
That shouldn't be an issue. The automated job checks that the notebooks execute without exceptions, but the job does not save the results or overwrite the documentation. So, we're still able to manually review any changes to the saved outputs.

LakshmanKishore commented:
I thought those warnings were coming due to some cache in the tensorflow I'm using, I tried doing so many things to remove those warnings, but couldn't

Also I had also updated this link https://github.com/shap/shap/pull/3393#discussion_r1399727020 from 4 to 5 and had planned to commit the same after removing those stderr. But in the mean time you made the changes. 
What do we do about it?

connortann commented:
If you're happy to raise a fresh PR with that change, it would be most welcome.
